easybuild/easyblocks/k/kokkos.py

```diff
@@ -0,0 +1,305 @@
+##
+# Copyright 2016-2025 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Kokkos, implemented as an easyblock
+
+@author: Jan Reuter (JSC)
+"""
+from easybuild.tools import LooseVersion
+
+import easybuild.tools.toolchain as toolchain
+from easybuild.base import fancylogger
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError, print_warning
+from easybuild.tools.config import build_option
+from easybuild.tools.modules import get_software_root
+from easybuild.tools.systemtools import AARCH64, get_cpu_architecture
+from easybuild.tools.toolchain.compiler import OPTARCH_GENERIC
+
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+
+KOKKOS_INTEL_PACKAGE_ARCH_LIST = [
+    'WSM',  # Intel Westmere CPU (SSE 4.2), removed in Kokkos 4.3
+    'SNB',  # Intel Sandy/Ivy Bridge CPU (AVX 1)
+    'HSW',  # Intel Haswell CPU (AVX 2)
+    'BDW',  # Intel Broadwell Xeon E-class CPU (AVX 2 + transactional mem)
+    'SKL',  # Intel Skylake Client CPU
+    'SKX',  # Intel Sky Lake Xeon E-class HPC CPU (AVX512 + transactional mem)
+    'ICL',  # Intel Ice Lake Client CPU (AVX512)
+    'ICX',  # Intel Ice Lake Xeon Server CPU (AVX512)
+    'SPR',  # Intel Sapphire Rapids Xeon Server CPU (AVX512)
+    'KNC',  # Intel Knights Corner Xeon Phi
+    'KNL',  # Intel Knights Landing Xeon Phi
+]
+
+KOKKOS_CPU_ARCH_LIST = [
+    'NATIVE'  # Local CPU architecture, available since LAMMPS 2Aug2023
```

Comment:
```suggestion
    'NATIVE'  # Local CPU architecture
```

Present before Kokkos 4.1.00